File: test-suite/src/main/java/helloworld/ServerBuilderListener.java
package helloworld;

// tag::imports[]
import io.grpc.ServerBuilder;
import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "ServerBuilderListenerTest")
// tag::clazz[]
@Singleton
public class ServerBuilderListener implements BeanCreatedEventListener<ServerBuilder<?>> {

    @Override
    public ServerBuilder<?> onCreated(BeanCreatedEvent<ServerBuilder<?>> event) {
        final ServerBuilder<?> builder = event.getBean();
        builder.maxInboundMessageSize(1024);
        return builder;
    }
}
// end::clazz[]

File: test-suite-python/src/main/python/helloworld/ServerBuilderListener.py
# tag::imports[]
from io.grpc import ServerBuilder
from jakarta.inject import Singleton
from micronaut.context.event import BeanCreatedEvent, BeanCreatedEventListener
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="ServerBuilderListenerTest")
# tag::clazz[]
@Singleton
class ServerBuilderListener(BeanCreatedEventListener[ServerBuilder]):

    def onCreated(self, event: BeanCreatedEvent[ServerBuilder]) -> ServerBuilder:
        builder = event.getBean()
        builder.maxInboundMessageSize(1024)
        return builder
# end::clazz[]
